hasher-matcher-actioner/hmalib/common/mappings.py

```diff
@@ -1,37 +1,206 @@
+       depending on SignalTypeMapping's internal state.
+    """
+
+    def __init__(self):
+        all_content_types = ToggleableContentTypeConfig.get_all()
+        all_signal_types = ToggleableSignalTypeConfig.get_all()
+
+        if len(all_content_types) + len(all_signal_types) == 0:
+            # Reasonably sure that we've never set values. So initialize with
+            # default values.
+            logger.info(
+                "First time calling HMASignalTypeMapping.__init__()? Creating default signal and content types.."
+            )
+            default_content_types, default_signal_types = self._init_configs()
+            super().__init__(default_content_types, default_signal_types)
+            return
+
+        enabled_content_types = list(
+            map(
+                lambda c: import_class(c.content_type_class),
+                filter(lambda ct: ct.enabled, all_content_types),
+            )
+        )
+        enabled_signal_types = list(
+            map(
+                lambda c: import_class(c.signal_type_class),
+                filter(lambda st: st.enabled, all_signal_types),
+            )
+        )
+        super().__init__(enabled_content_types, enabled_signal_types)
+
+    def _init_configs(
+        self,
+    ) -> t.Tuple[t.List[t.Type[ContentType]], t.List[t.Type[SignalType]]]:
+        """
+        Initialize configs such that a subsequent call returns
+        DEFAULT_SIGNAL_AND_CONTENT_TYPES.
+
+        WARNING: Not designed for distributed usage. If invoked simultaneously
+        by multiple workloads, this may error out because configs.create will be
+        called on the same object.
+        """
+        full_class_name = lambda c: f"{c.__module__}.{c.__name__}"
+
+        content_types = []
+        for _type in DEFAULT_SIGNAL_AND_CONTENT_TYPES.content_by_name.values():
+            content_config = ToggleableContentTypeConfig(
+                name=f"content_type:{full_class_name(_type)}",
+                content_type_class=full_class_name(_type),
+            )
+            create_config(content_config)
+            content_types.append(_type)
+
+        signal_types = []
+        for _type in DEFAULT_SIGNAL_AND_CONTENT_TYPES.signal_type_by_name.values():
+            signal_config = ToggleableSignalTypeConfig(
+                name=f"signal_type:{full_class_name(_type)}",
+                signal_type_class=full_class_name(_type),
+            )
+            create_config(signal_config)
+            signal_types.append(_type)
+
+        return (content_types, signal_types)
+
+    def get_signal_type(self, name: str) -> t.Optional[t.Type[SignalType]]:
+        return self.signal_type_by_name.get(name, None)
+
+    def get_content_type(self, name: str) -> t.Optional[t.Type[ContentType]]:
+        return self.content_by_name.get(name, None)
+
+    def get_signal_type_enforce(self, name: str) -> t.Type[SignalType]:
```

Comment:
(alt, ignorable): Another convention I've seen that is less verbose is just sticking `x` at the end. 
* `get` -> nullable
* `getx` -> non-nullable

Reply:
I thought the same, but "getx python" search results always point to graphics packages. 

If you feel strongly, this is just a search-and-replace. Can easily do. LMK.

Reply:
I don't - in my personal reviewing convention:

blocking: I feel strongly about this. Don't land without explicitly confirming it's okay
nit: I wouldn't write it this way, and I think it's sub-optimal, or it's a suboptimal but still functionally correct
ignorable: If you didn't change it, my feelings are not hurt :P 
alt: Here's another way to write this, but your way is still correct. Sometimes I'm not sure if it's better either. Implicitly ignorable
code golf: Here's another way to write this, but I did it mostly to see if this would work. Implicitly ignorable